src/sentry/utils/http.py

```diff
@@ -64,7 +64,7 @@ def is_same_domain(url1, url2):
 
 
 def get_origins(project=None):
-    if settings.SENTRY_ALLOW_ORIGIN == "*":
+    if settings.SENTRY_ALLOW_ORIGIN in ("*", None):
```

Comment:
Having `None` in this list and the default set to `None` seems like it might cause project key origin lists to behave incorrectly on self-hosted installs. Instead of reading the project option like they do today, this method would return `*` all the time.

Reply:
good catch - i will rework the logic here to just branch based on if project is set or not.

if a project is set, we never use SENTRY_ALLOW_ORIGIN, if it isnt, we only use it